### DISCUSSION TITLE
Table Compression Size with Many Columns

### DISCUSSION THREAD

**Discussion by keastrid** [nom-tam-fits/nom-tam-fits#595] [Q&A]
Hello again,

We recently ran into issues with tables only having 1000 columns (a limit of the TFORMn cards, it seems; maybe something BinaryTable should validate before attempting to write?).

To work around this, we've begun splitting the table into multiple HDUs. When I did so, I noticed that the compressed file size was ~8x the uncompressed table's, and when inspecting the files via fv I saw that it has increased the number of rows to match the number of columns. I'm assuming it's the difference in row counts that is causing this increase in file size, but funpack and nom.tam have no problem working with the file and when reading give the correct row count. Is this correct, or is there some other way to handle tables with many columns?

![image](https://github.com/nom-tam-fits/nom-tam-fits/assets/70925169/b900dc3e-eeff-4a16-9c04-ba3aefcae637)

[Tables.zip](https://github.com/nom-tam-fits/nom-tam-fits/files/14898848/LCO.many.zip)

**Comment by attipaci** [comment]
Hi @keastrid,

The FITS specification allows for maximum 999 columns. This is because many table keywords have a 5 character stem, which leaves maximum 3 digits as the column index. So, yes you have no choice but to split the table.

I look at the compressed tables more closely, to see why the columns and rows get mixed up. There is definitely something fishy with the rows = cols in the compressed table...

**Comment by attipaci** [comment]
Created issue #596
